Display name for theme

jrapoport opened the pull request:
Added “AWL_THEME_DISPLAYNAME” which corresponds to the display name for bundle to decouple the theme name (defacto id) from something that can be
localized & displayed to the user.

Commit:
Merge branch 'appwilldev/master'

Commit:
Display name

Added “AWL_THEME_DISPLAYNAME” which corresponds to the display name for
bundle to decouple the theme name (id) from something that can be
localized & displayed to the user.

Neeeo commented:
Why not to use `objectForKey:` ? Developer can get anything from this method.
Display name isn't essential for ThemeManager. Let's keep it simple.
In my actual project, I have custom request too, like theme icon. So I create subclass from ThemeManager, and add new method to it.

jrapoport commented:
I started off with an extension, but it thought that since it's optional, it's good practice / more portable / convenient just to standardize something & decouple it for localization, rather than making everyone who wants to localize have to subclass just to add the same 4 lines over & over.

Commit:
Add colorWithPatternImage support

Now if you specify as color as “#[ImageName]” it will search for the
“Foo” image using the usual methods & if found, return a color using
that image as a pattern. To change the alpha, use
“#[ImageName],[Alpha]” e.g. #Foo,0.3

Commit:
Merge branch 'appwilldev/master'

jrapoport commented:
whoops. meant to put 63c7e3f on a separate branch. sorry.

Neeeo commented:
For localization, i18n support is better than one display name string.
For end user experience, theme screenshot is more convenient.
